### PR TITLE
[POC] Report unknown subtype error for GCP rules

### PR DESCRIPTION
**Pull request by orouz**
### Summary of your changes

this POC adds a field to OPA (`errors`) which currently returns an error msg if the `input.subType` is unknown

in a similar way we can also account for missing evidence, which also fails silently.  

### related issues:
 - https://github.com/elastic/cloudbeat/issues/672 
 - https://github.com/elastic/cloudbeat/issues/1509

### CODE REVIEW

File: bundle/compliance/cis_gcp/rules/cis_1_4/rule.rego [hunk 1]
```diff
@@ -3,14 +3,25 @@ package compliance.cis_gcp.rules.cis_1_4
...
 
 	result := common.generate_result_without_expected(
 		common.calculate_result(audit.is_policy_not_managed_by_user),
-		input.resource.iam_policy.bindings[i].members,
+		evidence,
 	)
 }
+
+evidence if {
+	input.resource.iam_policy.zoo[i].members
```
Comment (orouz) [inline bundle/compliance/cis_gcp/rules/cis_1_4/rule.rego:21]: intentionally return undefined

File: bundle/compliance/main.rego [hunk 2]
```diff
@@ -37,3 +43,9 @@ findings = f {
 }
 
 metadata = common.metadata
+
+# reports unknown subtype
+errors contains error if {
+	not input.subType in gcptypes.subtypes
+	error := sprintf("unknown subtype: '%s'", [input.subType])
+}
```
Comment (orouz) [inline bundle/compliance/main.rego:51]: when a subtype is not in the `subtypes` set, this error comes up in the OPA result. meant to indicate developers they're using an unknown subtype

Screenshot 2023-08-01 at 21 37 10

File: bundle/compliance/policy/gcp/data_adapter.rego [hunk 3]
```diff
@@ -7,31 +7,3 @@ resource = input.resource.resource
...
-	input.subType == "gcp-bigquery-dataset"
-}
-
-is_bigquery_table {
-	input.subType == "gcp-bigquery-table"
-}
-
-is_compute_instance {
-	input.subType == "gcp-compute-instance"
-}
```
Comment (orouz) [inline bundle/compliance/policy/gcp/data_adapter.rego:37]: using a set removes the repetitiveness in this file, so we only use the relevant strings twice:
1 - defined in the set `subtypes`
2 - the rule itself by calling `common.is_gcp_subtype`

File: bundle/compliance/cis_gcp/rules/cis_1_15/rule.rego [hunk 4]
```diff
@@ -15,7 +15,7 @@ is_valid if {
 }
 
 finding = result if {
-	data_adapter.is_api_key
+	common.is_gcp_subtype("gcp-apikeys-key")
```
Comment (orouz) [inline bundle/compliance/cis_gcp/rules/cis_1_15/rule.rego:18]: the main difference here is that we don't introduce another layer of abstraction: `is_api_key` and instead define the string once in a set and use it whenever needed, this IMO makes it more explicit and less complicated

**Review by oren-zohar — APPROVED** [review]
LGTM. I'm not sure if you need to update it for the new resources types

**Comment by orouz** [comment]
going to update this but merge for 8.11 as it's not critical, just makes things nicer. hopefully we will use this for azure as well.

**Pull request closed without merging**